### PR TITLE
enhance parameter bridge enable options for topics

### PR DESCRIPTION
**Pull request by daisukes**
This enhancement enables parameter_bridge to control the bridge precisely with the following keys. Please see the detail in doc/parameter_bridge.rst

 - direction
 - transient_local
 - reliable
 - latch

I was struggling with tf_static which is a latched topic in ROS1 but cannot be seen in ROS2.
With specifying transient_local and reliable to true, it can be seen in ROS2 like a latched topic (of course need to use transient_local and reliable for the subscriber).

Signed-off-by: Daisuke Sato <daisukes@cmu.edu>

### CODE REVIEW

File: doc/parameter_bridge.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+--------------------
+
+- **topics**: <array of **topic**>
+- **topic**: <dict>
+
+  .. csv-table:: 
+   :header: "key", "required", "default", "description"
+   :widths: 20, 10, 10, 50
+
+   **topic_name**, YES, N/A, Name of the topic to bridge which is used in both ROS1/2
```
Comment (gonzodepedro) [inline doc/parameter_bridge.rst:60]: Should it be "topic" and not "topic_name"?

Comment (daisukes) [reply doc/parameter_bridge.rst]: Thank you for checking! I fixed it.

**Comment by gonzodepedro** [comment]
[Build Status](https://ci.ros2.org/job/ci_packaging_linux/353/)

File: README.md [hunk 2]
```diff
@@ -14,6 +14,8 @@ Note that before `rostopic echo` would work with bridged topics, a subscriber mu
 You can use the `--bridge-all-2to1-topics` option to bridge all ROS 2 topics to ROS 1 so that tools such as `rostopic list` and `rqt` will see the topics even if there are no matching ROS 1 subscribers.
 Run `ros2 run ros1_bridge dynamic_bridge -- --help` for more options.
 
+If you want to specify topics you want to bridge, run `ros2 run ros1_bridge parameter_bridge`. See [here](doc/parameter_bridge.rst) for more information.
```
[the remark this answers is not in the document]
Comment (dirk-thomas) [reply README.md:17]: The `parameter_bridge` also supports service which should be mentioned here.

**Comment by dirk-thomas** [comment]
@daisukes Friendly ping.

**Comment by daisukes** [comment]
@dirk-thomas Thank you for the reminder. I will work on this today.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (daisukes) [reply README.md:17]: fixed the description

File: doc/parameter_bridge.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
+parameter_bridge configuration
+==============================
+
+You can launch parameter_bridge with the following command::
+  
+    ros2 run ros1_bridge parameter_bridge
+
+Parameter bridge reads ``/topics`` parameter on ROS1 to make your bridge. Here is an example of configuration::
```
[the remark this answers is not in the document]
Comment (daisukes) [reply doc/parameter_bridge.rst:8]: https://github.com/ros2/ros1_bridge/blob/5094cb5accec5b7f4b6a2a7d52e74eb5ba44daf6/src/parameter_bridge.cpp#L61

It takes arguments for the names of the parameters. I add a description of the arguments.

File: src/bridge.cpp [hunk 4]
```diff
@@ -19,6 +19,15 @@
 
 namespace ros1_bridge
 {
+rmw_qos_profile_t
+get_qos(size_t queue_size, bool transient_local, bool reliable) {
```
[the remark this answers is not in the document]
Comment (daisukes) [reply src/bridge.cpp:23]: fixed

File: src/parameter_bridge.cpp [hunk 5]
```diff
@@ -74,24 +76,52 @@ int main(int argc, char * argv[])
...
+      std::string direction = static_cast<std::string>(topics[i]["direction"]);
+      bool transient_local = static_cast<bool>(topics[i]["transient_local"]);
+      bool reliable = static_cast<bool>(topics[i]["reliable"]);
+      bool latch = static_cast<bool>(topics[i]["latch"]);
+      
       if (!queue_size) {
         queue_size = 100;
       }
+      if (direction == "") {
+	direction = "bidirectional";
```
[the remark this answers is not in the document]
Comment (daisukes) [reply src/parameter_bridge.cpp:88]: fixed

**Comment by daisukes** [comment]
Do I need to fix the conflict too?

File: README.md [hunk 6]
```diff
@@ -14,6 +14,8 @@ Note that before `rostopic echo` would work with bridged topics, a subscriber mu
 You can use the `--bridge-all-2to1-topics` option to bridge all ROS 2 topics to ROS 1 so that tools such as `rostopic list` and `rqt` will see the topics even if there are no matching ROS 1 subscribers.
 Run `ros2 run ros1_bridge dynamic_bridge -- --help` for more options.
 
+If you want to specify messages/services you want to bridge, run `ros2 run ros1_bridge parameter_bridge`. See [here](doc/parameter_bridge.rst) for more information.
```
Comment (dirk-thomas) [inline README.md:17]: Nitpick: please follow the style guide for documentation - namely one sentence per line.

Same below.

Comment (daisukes) [reply README.md]: Fixed the style and update the description.

**Comment by dirk-thomas** [comment]
ros2/design#296 aims to provide a way to configure the QoS from the outside in a generic way. With that approach being the preferred choice I don't think a custom way to configure the bridge is desired.

**Comment by sloretz** [comment]
Linking updated configurable QoS proposal https://github.com/ros2/design/pull/300

**Comment by hidmic** [comment]
@daisukes does the added functionality satisfy your use case? If so, you may go ahead and close this.

**Comment by daisukes** [comment]
@hidmic, yes for ROS2 galactic. I may want to use `direction` and `latch` if I keep using the bridge, but I'm planning to port all my packages to ROS2 next year. Anyway, I will close this.

**Pull request closed without merging**